inst/rmarkdown/templates/oup/resources/template.tex

```diff
@@ -96,48 +122,76 @@
   \fi
 }
 
-\newcommand*\thanksbycode[1]{%
-  \ifcounter{FNCT@#1}
-    {\samethanks[\value{FNCT@#1}]}
-    {\thanks{\Ref{FN}{#1}}\newcounter{FNCT@#1}\setcounter{FNCT@#1}{\value{footnote}}}
+\newcommand*\addrlblbycode[1]{%
+  \ifcounter{ADDRLBL@#1}
+    {}
+    {\refstepcounter{addrcnt}\newcounter{ADDRLBL@#1}\setcounter{ADDRLBL@#1}{\value{addrcnt}}}%
+    \arabic{ADDRLBL@#1}%
+}
+
+\newcommand*\addrbycode[1]{%
+  \ifcounter{ADDR@#1}
+    {}
+    {\newcounter{ADDR@#1}%
+     \address[\addrlblbycode{#1}]{\MyRef{ADDRTXT}{#1}}}%
+}
+
+\newcommand*\corresplblbycode[1]{%
+  \ifcounter{CORRESPLBL@#1}
+    {}
+    {\refstepcounter{correspcnt}\newcounter{CORRESPLBL@#1}\setcounter{CORRESPLBL@#1}{\value{correspcnt}}}%
+    \fnsymbol{CORRESPLBL@#1}%
+}
+
+\newcommand*\correspbycode[1]{%
+  \ifcounter{CORRESP@#1}
+    {}
+    {\newcounter{CORRESP@#1}%
+     \corresp[\corresplblbycode{#1}]{\MyRef{CORRESPTXT}{#1}}}%
 }
 
+\makeatother
+
+% Add missing \city command mentioned in documentation but absent from cls
+\providecommand\city[1]{#1}
+
 % Create labels for Addresses if the are given in Elsevier format
-$if(address)$
 $for(address)$
-$if(address.code)$
-\Newlabel{ADR}{$address.code$}{$address.address$}
-$endif$
-$if(address.id)$
-\Newlabel{ADR}{$address.id$}{$address.address$}
-$endif$
+ $if(address.code)$
+  \MyNewLabel{ADDRTXT}{$address.code$}{%
+ $elseif(address.id)$
+  \MyNewLabel{ADDRTXT}{$address.id$}{%
+ $endif$
+ $if(address.address)$$address.address$$else$%
+  $if(address.department)$\orgdiv{$address.department$}, $endif$%
+  $if(address.organization)$\orgname{$address.organization$}, $endif$%
+  $if(address.street)$\orgaddress{%
+   $elseif(address.city)$\orgaddress{%
+   $elseif(address.postcode)$\orgaddress{%
+   $elseif(address.state)$\orgaddress{%
+   $elseif(address.country)$\orgaddress{%
+  $endif$%
+  $if(address.street)$\street{$address.street$}, $endif$%
+  $if(address.postcode)$\postcode{$address.postcode$}, $endif$%
+  $if(address.city)$\city{$address.city$}, $endif$%
+  $if(address.state)$\state{$address.state$}, $endif$%
+  $if(address.country)$\country{$address.country$}$endif$%
+  $if(address.street)$}%
+   $elseif(address.city)$}%
+   $elseif(address.postcode)$}%
+   $elseif(address.state)$}%
+   $elseif(address.country)$}%
+  $endif$%
+ $endif$%
+ }
 $endfor$
-$endif$
 
-% Create labels for Footnotes if the are given in Elsevier format
-$if(footnote)$
+% Create labels for Footnotes if they are given in Elsevier format
 $for(footnote)$
 $if(footnote.code)$
-\Newlabel{FN}{$footnote.code$}{$footnote.text$}
+\MyNewLabel{CORRESPTXT}{$footnote.code$}{$footnote.text$}
 $endif$
 $endfor$
-$endif$
-
-% Part for setting citation format package: natbib
-$if(natbib)$
-\usepackage{natbib}
-\bibliographystyle{$if(biblio-style)$$biblio-style$$else$plainnat$endif$}
-$endif$
```

Comment:
Why remove this ? 

We still need `\usepackage{natbib}`, don't we ?

Reply:
This has been moved into the CLS file as a required package. 

```latex
\RequirePackage{natbib}
```

Reply:
Oh ok ! 

So `natbib` is really the only possible citation package that you can use. You could not use `pandoc` citation at all with this template ? Is that right ? 
Because even if you don't want to use natbib (`citation_package: default`), the package `natbib` will be required. (and maybe could lead to issues ?)

If so, we need to force `natbib` in the format function then.

Reply:
Well... With my template, even if `natbib` is a required package, pandoc citations using `.csl` files will work just fine. I thought initially that I would force the user to use `natbib`, but if `.csl` citations can also be made to work, why not. I think one of the reasons that more OUP journals have not adapted the new format is specific citation styles, so if we can use pandoc to produce those formats, then perhaps the journals will be happy to accept the articles...